assimilation_code/modules/io/io_filenames_mod.f90

```diff
@@ -690,21 +717,23 @@ end subroutine check_attribute_value_int
 !> check r4 values are the same
 
 
-subroutine check_attribute_value_r4(ncFile, filename, ncVarID, att_string, spvalR4)
+subroutine check_attribute_value_r4(ncFile, filename, ncVarID, att_string, att_valR4)
```

Comment:
i honestly don't believe that we have to have int and r4 and r8 versions of all these routines.  i was hoping they'd all go away so i haven't said much before.  but we use the default fortran conversion routines to go from what's in the netcdf files to our real(r8) arrays, so that should do the same for comparisons/tests of missing values or fill values using the fortran intrinsics.